### PR TITLE
Code modifications in order to change the DNS word creation for Spot-…

### PR DESCRIPTION
**Pull request by brandon-edwards**
…ML. See JIRA issue SPOT-158.

### CODE REVIEW

**Comment by brandon-edwards** [comment]
unit tests passing and test cluster run passes (I ran it on prod in /home/duxbury/test_ml with ml_ops_oni.sh which sources /etc/spot_oni.conf)

File: spot-ml/src/main/scala/org/apache/spot/dns/DNSWordCreation.scala [hunk 1]
```diff
@@ -90,8 +79,8 @@ class DNSWordCreation(frameLengthCuts: Array[Double],
     * @return The word representation of the DNS entry.
     */
 
-  def dnsWord(timeStamp: String,
-              unixTimeStamp: Long,
+  def dnsWord(timestamp: String,
```
Comment (NathanSegerlind) [inline spot-ml/src/main/scala/org/apache/spot/dns/DNSWordCreation.scala]: the English word is "time stamp" so timeStamp is the correct rendering in camel case
http://www.dictionary.com/browse/time--stamp?s=t

**Comment by lujacab** [comment]
+1

File: spot-ml/src/main/scala/org/apache/spot/utilities/Quantiles.scala [hunk 2]
```diff
@@ -1,124 +0,0 @@
...
-    }
-    valueCountPairs.unpersist()
-    // normalize counts by  total number entries in the data to obtain the ecdf
-    valueCountLEQPairs.map({case (value, countToLeftOfValue) => (value, countToLeftOfValue / totalCount)})
-  }
-
-  def bin(value: Double, cuts: Array[Double]) : Int = {
-    cuts.indexWhere(cut => value <= cut)
-  }
-}
```
Comment (rabarona) [inline spot-ml/src/main/scala/org/apache/spot/utilities/Quantiles.scala]: Bye Quantiles.scala

File: spot-ml/src/main/scala/org/apache/spot/dns/DNSWordCreation.scala [hunk 3]
```diff
@@ -104,11 +93,11 @@ class DNSWordCreation(frameLengthCuts: Array[Double],
         extractDomainInfo(queryName, topDomainsBC, userDomain)
 
       Seq(topDomain,
-        Quantiles.bin(frameLength.toDouble, frameLengthCuts),
-        Quantiles.bin(unixTimeStamp.toDouble, timeCuts),
-        Quantiles.bin(subdomainLength.toDouble, subdomainLengthCuts),
-        Quantiles.bin(subdomainEntropy, entropyCuts),
-        Quantiles.bin(numPeriods.toDouble, numberPeriodsCuts),
+        MathUtils.logBaseXInt(frameLength.toDouble, 2),
```
Comment (lujangus) [inline spot-ml/src/main/scala/org/apache/spot/dns/DNSWordCreation.scala]: Good, in the future we could assign a base other than 2 to play around and see if we are able to improve performance.

**Pull request merged**